Lux/src/life.cpp

```diff
@@ -117,36 +95,27 @@ template< class T > bool life< T >::operator () ( buffer_pair< T >& buf, const f
  *  http://www.bitstorm.org/gameoflife/lexicon/
  */
 
-template< class T > bool life< T >::operator () ( buffer_pair< T >& buf, const float& t ) { 
-    T white, black;
-    ::white( white ); ::black( black );
-    MOORE_NEIGHBORHOOD_BEGIN
-            // implement game of life rule 
-            // ***************************
-            
-            int count = 0;
-            if( *ul == white ) count++;  if( *um == white ) count++;  if( *ur == white ) count++; 
-            if( *ml == white ) count++;                               if( *mr == white ) count++; 
-            if( *dl == white ) count++;  if( *dm == white ) count++;  if( *dr == white ) count++;
-
-            // count = *ul + *um + *ur + *ml + *mr + *dl + *dm + *dr;
-            if( *mm == white )
-            {
-                // case where center cell is filled
-                if(count == 2 | count == 3) *result = white; else *result = black;
-            }
-            else
-            {
-                // case where center cell is empty
-                if(count == 3) *result = white; else *result = black;
-            }   
-
-            // ***************************
-            // end rule implementation
-
-    MOORE_NEIGHBORHOOD_END
+template<class T> void life<T>::operator()(const std::vector<T>& neighbors, std::vector<T>& result) {    
+    int count = 0;
+    for( int i=0; i<4; i++ ) {
+        if( neighbors[ i ] == on ) count++;
```

Comment:
i would write this without an if like
```
count += (neighbors[ i ] == on);
```
can also combine loops probably i think optimizer can still make it effiicent;
```
count += (i != 4) * (neighbors[ i ] == on);
```

Reply:
can also just add up 0-9 then subtract neighbors[4] at the end. seems like less code than two loops. all ways work here though i guess

Reply:
Thanks for looking at this! I'll make at least the first change for the next PR.